core/src/main/java/org/apache/carbondata/core/writer/CarbonIndexFileMergeWriter.java

```diff
@@ -95,22 +97,20 @@ private String mergeCarbonIndexFilesOfSegment(String segmentId,
               indexFilesInPartition.add(indexCarbonFile);
             }
           }
-          indexFiles = indexFilesInPartition.toArray(new CarbonFile[indexFilesInPartition.size()]);
+          indexFiles = indexFilesInPartition;
         } else {
-          indexFiles = indexCarbonFiles.toArray(new CarbonFile[indexCarbonFiles.size()]);
+          indexFiles = indexCarbonFiles;
         }
+      }
+      if (indexFiles.isEmpty() || indexFileNamesTobeAdded != null) {
+        return writeMergeIndexFileBasedOnSegmentFolder(indexFileNamesTobeAdded,
+            readFileFooterFromCarbonDataFile, segmentPath,
+            indexFiles.toArray(new CarbonFile[indexFiles.size()]), segmentId);
       } else {
-        indexFiles =
+        CarbonFile[] indexFilesFromFile =
```

Comment:
you should remove this and add in if condition when segment file is null, please check

```
CarbonFile[] indexFilesFromFile =
            SegmentIndexFileStore.getCarbonIndexFiles(segmentPath, FileFactory.getConfiguration());
```

Reply:
done